### PR TITLE
Add the buildpack

### LINKED ISSUE #11: Create a buildpack Dockerfile
**Description**

Create a buildpack Dockerfile with all the tooling needed for building and testing Kyma components. This buildpack will be used by all the ProwJobs later on.  

**Acceptance Criteria**

- Create a buildpack as a Dockerfile with all the tooling needed for building and testing Kyma components.
- It must include Golang and Node.js tooling as well as other basics needed such as git and gcloud.
- It must be able to build and push Docker images.

### PR DESCRIPTION
**Pull request by suleymanakbas91**
**Description**

Changes proposed in this pull request:

- Create a buildpack as a Dockerfile with all the tooling needed for building and testing Kyma components.
- It is using Docker in Docker to be able to build and push Docker images.

**Related issue(s)**

Resolves #11

### CODE REVIEW

File: buildpack/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        --usage-reporting=false && \
+    gcloud components install alpha beta kubectl && \
+    gcloud info 
+
+#
+# BEGIN: DOCKER IN DOCKER SETUP
+#
+
+# Install Docker deps
+RUN apt-get update && apt-get install -y --no-install-recommends \
```
Comment (mwieczorek) [inline buildpack/Dockerfile]: if we want to make the image smaller then please run at the end:
```
    && apt-get autoremove -y \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
```
of course in every apt-get install (maybe we can also combine all those apt-get install into one to have less image layers?)

File: buildpack/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,94 @@
+FROM golang:1.11.0
```
[the remark this answers is not in the document]
Comment (suleymanakbas91) [reply buildpack/Dockerfile]: if we split them, we will need to duplicate the code for all the other tools to have them on both (like docker in docker setup).

[the remark this answers is not in the document]
Comment (mwieczorek) [reply buildpack/Dockerfile]: maybe we can have a base image with all required tools and 2 additional images, one for golang, one for node?

File: buildpack/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+## NODE.JS
+RUN curl -sL https://deb.nodesource.com/setup_10.x | bash -
+RUN apt-get update && apt-get install -y nodejs uidmap libseccomp-dev
+
+RUN nodejs -v
+
+## GOLANG
+# install dep
+RUN go get -u github.com/golang/dep/cmd/dep
```
[the remark this answers is not in the document]
Comment (suleymanakbas91) [reply buildpack/Dockerfile]: I've done it for the ones I could find a way to do.

File: buildpack/Dockerfile (hunk 1 above)
Comment (suleymanakbas91) [reply buildpack/Dockerfile]: Done

File: buildpack/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (mwieczorek) [reply buildpack/Dockerfile]: @suleymanakbas91  dep is still installed from master, not from 0.4.1 tag?

**Comment by suleymanakbas91** [comment]
Closing this as our approach has shifted to have a job per each component instead of a single job for all the components.

**Pull request closed without merging**